grails-app/services/com/unifina/service/CommunitySecretService.groovy

```diff
@@ -33,7 +34,12 @@ class CommunitySecretService {
 		if (result == null) {
 			throw new NotFoundException("Community secret not found")
 		}
-		result.name = cmd.name
+		if (cmd.name) {
+			result.name = cmd.name
+		}
+		if (cmd.secret) {
+			throw new BadRequestException("Can't change the secret, please create a new community secret")
```

Comment:
Needs Swagger update

Reply:
Swagger is updated now. That parameter is not in the spec, so it returns the 400 (which is already in the spec).